Fix `ReferenceError: weakly-referenced object no longer exists` in cycle detector

zdevito opened the pull request:
Summary: weakref.proxy objects will throw errors when they re dead.  We just do not bother visulaizing them. They are weak, so they aren't relevant to cycles anyway.

Differential Revision: D69270429

tianfengfrank commented on torch/utils/viz/_cycles.py:
```diff
@@ -391,8 +390,61 @@ def to_dot(nodes):
 let dot = $DOT
-let image = Viz(dot, {format: 'svg'});
-document.getElementById('main').innerHTML = image
+let image = Viz(dot, {format: 'svg', 'totalMemory': 1024*1024*1024});
```
nit: should we define the 1024*1024*1024 as constants?

zdevito replied:
I think its fine since it is only used in this one place.

tianfengfrank reviewed (approved):
Overall LGTM! Thanks for fixing the issue!

zdevito commented:
@pytorchbot rebase

zdevito commented:
@pytorchbot merge

zdevito commented:
@pytorchbot merge

zdevito committed:
Fix `ReferenceError: weakly-referenced object no longer exists` in cycle detector (#146922)

Summary:

weakref.proxy objects will throw errors when they re dead.  We just do not bother visulaizing them. They are weak, so they aren't relevant to cycles anyway.

Test Plan:
run _cycles.py on an example of a cycle

Tested with mast job run: aps-fbfmv4_0105_mem_RCD_5-67146bd290
https://www.internalfb.com/mlhub/pipelines/runs/mast/aps-fbfmv4_0105_mem_RCD_5-67146bd290?job_attempt=1&version=0&tab=summary&env=PRODUCTION

Reviewed By: Chillee, tianfengfrank

Differential Revision: D69270429
